src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -568,7 +568,7 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
                 result.WasCompilerGenerated = true;
                 diagnostics.AddRange(bag);
 
-                if (bag.HasAnyErrors())
+                if (bag.HasAnyResolvedErrors())
                 {
```

Comment:
We should not be looking in the bag for errors to decide if we've found a suitable deconstruct method or not.